src/runtime/pkg/kata-monitor/cri.go

```diff
@@ -149,61 +146,12 @@ func (km *KataMonitor) getSandboxes(sandboxMap map[string]bool) (map[string]bool
 			continue
 		}
 
-		request := &pb.PodSandboxStatusRequest{
-			PodSandboxId: pod.Id,
-			Verbose:      true,
-		}
-
-		r, err := runtimeClient.PodSandboxStatus(context.Background(), request)
-		if err != nil {
-			return newMap, err
-		}
-
-		lowRuntime := ""
-		var res map[string]interface{}
-		if err := json.Unmarshal([]byte(r.Info["info"]), &res); err != nil {
-			monitorLog.WithError(err).WithField("pod", r).Error("failed to Unmarshal pod info")
-			continue
-		} else {
-			monitorLog.WithField("pod info", res).Debug("")
-
-			// get low level container runtime
-			// containerd stores the pod runtime in "/runtimeType" while CRI-O stores it the
-			// io.kubernetes.cri-o.RuntimeHandler annotation: check for both.
-			const (
-				containerdRuntimeMarker = "/runtimeType"
-				crioRuntimeMarker       = "/runtimeSpec/annotations/io.kubernetes.cri-o.RuntimeHandler"
-			)
-			keys := []string{containerdRuntimeMarker, crioRuntimeMarker}
-			for _, key := range keys {
-				pointer, _ := gojsonpointer.NewJsonPointer(key)
-				rt, _, _ := pointer.Get(res)
-				if rt != nil {
-					if str, ok := rt.(string); ok {
-						lowRuntime = str
-						break
-					}
-				}
-			}
-		}
-
-		// If lowRuntime is empty something changed in containerd/CRI-O or we are dealing with an unknown container engine.
-		// Safest options is to add the POD in the list: we will be able to connect to the shim to retrieve the actual info
-		// only for kata PODs.
-		if lowRuntime == "" {
-			monitorLog.WithField("pod", r).Warning("unable to retrieve the runtime type")
-			newMap[pod.Id] = true
-			continue
-		}
-
+		newMap[pod.Id] = checkSandboxFSExists(pod.Id)
```

Comment:
I would add a comment explaining what is going on here. Specifically, that you check the sandbox FS as a side channel to figure out if this is kata.

Reply:
Yeah, otherwise it would be a bit obscure.. added!